presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -233,6 +233,8 @@ void updateTaskStatus(TaskStatus newValue)
         });
 
         if (taskMismatch.get()) {
+            // This will issue a DELETE for the task to the worker.
+            // It is desirable because a task was created by the previous request.
             onFail.accept(new PrestoException(REMOTE_TASK_MISMATCH, format("%s (%s)", REMOTE_TASK_MISMATCH_ERROR, HostAddress.fromUri(getTaskStatus().getSelf()))));
```

Comment:
I think we also need to force the task into a FAILED state on the coordinator.  I am concerned, we could leave the query in an incomplete state.  Think this code should do the trick:
```java
            PrestoException prestoException = new PrestoException(REMOTE_TASK_MISMATCH, format("%s (%s)", REMOTE_TASK_MISMATCH_ERROR, HostAddress.fromUri(getTaskStatus().getSelf())));
            while (true) {
                TaskStatus currentStatus = this.taskStatus.get();
                if (currentStatus.getState().isDone()) {
                    return;
                }

                TaskStatus failedStatus = new TaskStatus(
                        currentStatus.getTaskId(),
                        currentStatus.getTaskInstanceId(),
                        currentStatus.getVersion() + 1,
                        FAILED,
                        currentStatus.getSelf(),
                        ImmutableList.of(toFailure(prestoException)),
                        0,
                        0,
                        new DataSize(0, BYTE));

                if (taskStatus.compareAndSet(currentStatus, failedStatus)) {
                    onFail.accept(prestoException);
                    return;
                }
            }
```